orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy

```diff
@@ -64,58 +66,118 @@ class FindImageFromClusterTask extends AbstractCloudProviderAwareTask implements
+      config.resolveMissingLocations = defaultResolveMissingLocations
     }
 
-    def onlyEnabled = stage.context.onlyEnabled == null ? true : (Boolean.valueOf(stage.context.onlyEnabled.toString()))
-    def selectionStrat = SelectionStrategy.valueOf(stage.context.selectionStrategy?.toString() ?: "NEWEST")
+    List<Location> missingLocations = []
 
-    Map<Location, Map<String, Object>> imageSummaries = requiredLocations.collectEntries { location ->
+    Set<String> imageNames = []
+    Map<Location, String> imageIds = [:]
+
+    Map<Location, Map<String, Object>> imageSummaries = config.requiredLocations.collectEntries { location ->
       try {
         def lookupResults = oortService.getServerGroupSummary(
-          app,
+          config.application,
           account,
-          cluster,
+          config.cluster,
           cloudProvider,
           location.value,
-          selectionStrat.toString(),
+          config.selectionStrategy.toString(),
           SUMMARY_TYPE,
-          onlyEnabled.toString())
+          config.onlyEnabled.toString())
+        imageNames << lookupResults.imageName
+        imageIds[location] = lookupResults.imageId
         return [(location): lookupResults]
       } catch (RetrofitError e) {
         if (e.response.status == 404) {
-          def message = "Could not find cluster '$cluster' for '$account' in '$location.value'."
+          final Map reason
           try {
-            Map reason = objectMapper.readValue(e.response.body.in(), new TypeReference<Map<String, Object>>() {})
-            if (reason.error.contains("target.fail.strategy")){
-              message = "Multiple possible server groups present in ${location.value}"
-            }
-          } catch (Exception ignored) {}
-          throw new IllegalStateException(message)
+            reason = objectMapper.readValue(e.response.body.in(), new TypeReference<Map<String, Object>>() {})
+          } catch (Exception ex) {
+            throw new IllegalStateException("Unexpected response from API")
+          }
+          if (reason.error?.contains("target.fail.strategy")){
+            throw new IllegalStateException("Multiple possible server groups present in ${location.value}")
+          }
+          if (config.resolveMissingLocations) {
+            missingLocations << location
+            return [(location): null]
+          }
+
+          throw new IllegalStateException("Could not find cluster '$config.cluster' for '$account' in '$location.value'.")
         }
         throw e
       }
     }
 
+    if (missingLocations) {
+      if (imageNames.size() != 1) {
+        throw new IllegalStateException("Request to resolve images for missing ${config.requiredLocations.first().pluralType()} requires exactly one image. (Found ${imageNames})")
+      }
+
+      def searchImage = imageIds.entrySet().first()
+
+      def deploymentDetailTemplate = imageSummaries.find { k, v -> v != null}.value
```

Comment:
I don't get this, doesn't it amount to the same thing as `def deploymentDetailTemplate = null`?